engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/node/ForeignFunctionCallNode.java

```diff
@@ -10,5 +13,6 @@ public abstract class ForeignFunctionCallNode extends Node {
    * @param arguments the arguments to pass to the foreign function
    * @return the result of executing the foreign function
    */
-  public abstract Object execute(Object[] arguments);
+  public abstract Object execute(Object[] arguments)
+      throws UnsupportedTypeException, ArityException, UnsupportedMessageException;
```

Comment:
Can't we handle all these exceptions in `PyForeignNode`? And if not, is it correct that we "cast" them to `RuntimeException` and let them propagate further? It seems to me that if some of these exception happen in `PyForeignNode`, they are rather internal because we have a full control over how the python code snippets there are run.

So my suggestion here is to either throw something like `IllegalStateException` in `ForeignEvalNode.execute`, or to handle these exceptions in `PyForeignNode`.

See https://github.com/enso-org/enso/blob/4b9c91626ebaf878fa5d9868a15177826db166ab/engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/node/RForeignNode.java#L24 for example - there we throw `IllegalStateException` as well.

Reply:
`IllegalStateException` is wrong, in my opinion. That signals failure in the engine and crashes the engine. Moreover it cannot be constructed on fast path - e.g. [RForeignNode is certainly wrong](https://github.com/enso-org/enso/blob/4b9c91626ebaf878fa5d9868a15177826db166ab/engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/node/RForeignNode.java#L24).

> they are rather internal because we have a full control over how the python code snippets

I am not sure about that. I assumed malformed input could generate one of those exceptions.